notebooks/09_Monitoring/src/07_01_Iris_Pipeline_Monitoring/model.py

```diff
@@ -0,0 +1,216 @@
+
+
+def preprocess(args):
+    iris = pd.read_csv(args.input_url, names=args.input_column_names)
+    array = iris.values
+    X, y = array[:, 0:4], np.where(array[:, 4] == 'Iris-setosa', 1, 0)
+    X_train, X_test, y_train, y_test = model_selection.train_test_split(X, y, test_size=args.test_size,
+                                                                        random_state=args.train_random_seed)
+
+    return X_train, X_test, y_train, y_test
+
+
+def sklearn_training(X_train, y_train):
+    model = LogisticRegression()
+    model.fit(X_train, y_train)
+
+    return model
+
+
+def inference(model, X_test, y_test):
+    y_pred = model.predict(X_test)
+    logging.info("Trained Model's evaluation score: {}".format(model.score(X_test, y_test)))
+    df = pd.concat([pd.DataFrame(y_test, columns=['target']), pd.DataFrame(y_pred, columns=['predicted'])], axis=1)
+    vocab = list(df['target'].unique())
+    cm = confusion_matrix(df['target'], df['predicted'], labels=vocab)
+    data = []
+    for target_index, target_row in enumerate(cm):
+        for predicted_index, count in enumerate(target_row):
+            data.append((vocab[target_index], vocab[predicted_index], count))
+    df_cm = pd.DataFrame(data, columns=['target', 'predicted', 'count'])
+    cm_file = os.path.join('/', 'confusion_matrix.csv')
+    with file_io.FileIO(cm_file, 'w') as f:
+        df_cm.to_csv(f, columns=['target', 'predicted', 'count'], header=False, index=False)
+
+    y_score = model.decision_function(X_test)
+    # Compute ROC curve and ROC area for each class
+    fpr, tpr, thresholds = roc_curve(y_test, y_score)
+    df_roc = pd.DataFrame({'fpr': fpr, 'tpr': tpr, 'thresholds': thresholds})
+    roc_file = os.path.join('/', 'roc.csv')
+    with file_io.FileIO(roc_file, 'w') as f:
+        df_roc.to_csv(f, columns=['fpr', 'tpr', 'thresholds'], header=False, index=False)
+
+    return cm_file, roc_file, vocab
+
+
+def tf_training(args, X_train, y_train, X_test, y_test):
+    model = tf.keras.Sequential([
+        tf.keras.layers.Dense(10, activation='relu', input_shape=(4,)),  # input shape required
+        tf.keras.layers.Dense(32, activation='relu'),
+        tf.keras.layers.Dense(64, activation='relu'),
+        tf.keras.layers.Dense(128, activation='relu'),
+        tf.keras.layers.Dense(2)
+    ])
+
+    model.compile(optimizer=args.tf_nn_optimizer,
+                  loss=args.tf_nn_loss_func,
+                  metrics=[args.tf_nn_metrics])
+
+    time_hash = datetime.datetime.now().strftime("%Y%m%d-%H%M%S")
+    log_dir = "/logs/fit/" + time_hash
+    tensorboard_callback = tf.keras.callbacks.TensorBoard(log_dir=log_dir)
+
+    model.fit(x=X_train,
+              y=y_train,
+              epochs=args.tf_nn_epoch,
+              validation_data=(X_test, y_test),
+              callbacks=[tensorboard_callback])
+
+    return model, log_dir, time_hash
+
+
+def upload_s3(args, cm_file, roc_file ,log_dir, time_hash):
+    AWS_REGION = args.aws_region
```

Comment:
Create Bucket similarly like AWS_REGIOn

Reply:
Okay, I put `bucket = args.s3_bucket` in the `try except` code block, will fix it.